### PR TITLE
KK-49 | Prepopulate fields on frontpage form is user has submitted it before

### PR DESCRIPTION
**Pull request by badrange**

### CODE REVIEW

File: src/domain/home/form/HomePreliminaryForm.tsx [hunk 1]
```diff
@@ -161,9 +143,13 @@ const actions = {
   setFormValues,
 };
 
+const mapStateToProps = (state: StoreState) => ({
+  stateFormValues: state.registration.formValues,
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryForm.tsx]: should be named `initialFormValues`

File: src/domain/home/form/HomePreliminaryFormUtils.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { get } from 'lodash';
+
+import { RegistrationFormValues } from '../../registration/types/RegistrationTypes';
+import { DEFAULT_DATE_FORMAT } from '../../../common/time/TimeConstants';
+import { newMoment } from '../../../common/time/utils';
+import { HomeFormValues } from './types/FormTypes';
+
+export const convertFormValues = (
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: function name should be getInitialFormValues

Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: js-doc missing
(Function documentation) what did function do, params, returns blabla

File: src/domain/home/form/HomePreliminaryFormUtils.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  stateFormValues: RegistrationFormValues
+): HomeFormValues => {
+  if (get(stateFormValues, 'child.birthDay')) {
+    const day = newMoment(stateFormValues.child.birthday, DEFAULT_DATE_FORMAT);
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: should be name `birthdayMoment`

File: src/domain/home/form/HomePreliminaryFormUtils.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    return {
+      child: {
+        birthday: {
+          day: day.date(),
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: birthdayMoment.get('date')

File: src/domain/home/form/HomePreliminaryFormUtils.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          month: day.month() + 1,
+          year: day.year(),
+        },
+        homeCity: stateFormValues.child.homeCity,
+      },
+      verifyInformation: stateFormValues.verifyInformation,
+    };
+  } else {
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: remove else (not needed)
return the default object

File: src/domain/home/form/types/FormTypes.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export interface HomeFormValues {
```
Comment (midori0507) [inline src/domain/home/form/types/FormTypes.ts]: file name should be HomeFormTypes

Comment (midori0507) [reply src/domain/home/form/types/FormTypes.ts]: for intellisense

File: src/domain/home/form/HomePreliminaryForm.tsx (hunk 1 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryForm.tsx]: But they are not initial, they are from state..

File: src/domain/home/form/HomePreliminaryFormUtils.ts (hunk 2 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryFormUtils.ts]: It feels stupid to send initialFormValues to a getInitialFormValues function..

File: src/domain/home/form/types/FormTypes.ts (hunk 6 above)
Comment (badrange) [reply src/domain/home/form/types/FormTypes.ts]: Hah, funny.

File: src/domain/home/form/HomePreliminaryForm.tsx (hunk 1 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryForm.tsx]: ..and I thought it would be confusing to use the same variable name as a "default" object.

Comment (midori0507) [reply src/domain/home/form/HomePreliminaryForm.tsx]: uhm that make sense

File: src/domain/home/form/HomePreliminaryFormUtils.ts (hunk 2 above)
Comment (midori0507) [reply src/domain/home/form/HomePreliminaryFormUtils.ts]: 😆

File: src/domain/home/form/HomePreliminaryFormUtils.ts (hunk 3 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryFormUtils.ts]: Good point!

File: src/domain/home/form/HomePreliminaryFormUtils.ts (hunk 2 above)
Comment (badrange) [reply src/domain/home/form/HomePreliminaryFormUtils.ts]: yeah!

File: src/domain/home/form/types/FormTypes.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+export interface HomeFormTypes {
```
Comment (midori0507) [inline src/domain/home/form/types/FormTypes.ts]: well the interface name should stay as it is before you move
`HomeFormValues`

But the filename shoud be HomeFormTypes

File: src/domain/home/form/HomePreliminaryFormUtils.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { get } from 'lodash';
+
+import { RegistrationFormValues } from '../../registration/types/RegistrationTypes';
+import { DEFAULT_DATE_FORMAT } from '../../../common/time/TimeConstants';
+import { newMoment } from '../../../common/time/utils';
+import { HomeFormTypes } from './types/FormTypes';
+
+/**
+ * Convert FormValues fetched from state to initialValues used by formi
```
Comment (midori0507) [inline src/domain/home/form/HomePreliminaryFormUtils.ts]: *formik

**Pull request merged**